lib/Reader/readerir.cpp

```diff
@@ -2137,6 +2138,21 @@ uint32_t GenIR::addArrayFields(std::vector<llvm::Type *> &Fields,
     FieldByteSize += DataLayout->getTypeSizeInBits(ArrayPadTy) / 8;
   }
 
+  // For multidimensional arrays there are arrays of dimension lengths and
+  // lower bounds. Note that single-dimensional arrays with a non-zero lower
+  // bound have the same layout as single-dimensional arrays with a zero
+  // lower bound.
```

Comment:
Might be worth describing how the nonzero lower bound is handled for the rank 1 case.